app/main.py

```diff
@@ -1,6 +1,21 @@
-from typing import Callable
+from typing import Callable, Any, Dict, Tuple, Hashable
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    cache_store: Dict[Callable, Dict[Tuple[Hashable, ...], Any]] = {}
+
+    def wrapper(*args, **kwargs) -> Any:
+        if func not in cache_store:
+            cache_store[func] = {}
+        key = (args, frozenset(kwargs.items()))
```

Comment:
Using `frozenset` to store keyword arguments in the cache key can lead to issues when the order of keyword arguments is different but the values are the same. Consider using a sorted tuple of items instead: `tuple(sorted(kwargs.items()))`.